instagram/src/components/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+const withAuthenticate = First => Second =>
+  class extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        login: false
+      };
+    }
+    componentDidMount() {
+      if (!localStorage.getItem("user")) {
+        this.setState({ login: false });
+      } else {
+        this.setState({ login: true });
+      }
+    }
+
+    render() {
+      console.log("Login", this.state.login);
+      return this.state.login === false ? (
+        <First login={this.state.login} />
+      ) : (
+        <Second />
+      );
+    }
+  };
+export default withAuthenticate;
```

Comment:
Excellent! Neatly written HOC!

To make the code more readable, you can specify the arguments as follow:
```
const withAuthenticate = Login => PostPage =>
```
and rename the `First` and `Second` component call to `Login` and `PostPage` respectively.